src/goCheck.ts

```diff
@@ -46,6 +51,66 @@ export function notifyIfGeneratedFile(this: void, e: vscode.TextDocumentChangeEv
 	}
 }
 
+function notifyOnBuildTagMismatch(ctx: any, e: vscode.TextDocumentChangeEvent) {
+	if (ctx.globalState.get('ignoreBuildTagMismatchWarning') === true) {
+		return;
+	}
+
+	const fileBuildTags = [];
+	for (let i = 0; i < e.document.lineCount; i++) {
+		const line = e.document.lineAt(i);
+		if (line.isEmptyOrWhitespace) {
+			break;
+		}
+		const bldTagLine = line.text.match(/^\/\/\s*\+build\s+(.+)$/);
+		if (bldTagLine) {
+			if (i === e.contentChanges[0].range.start.line) {
+				return;  // editing +build tag line
+			}
+			fileBuildTags.push(bldTagLine[1]);
+		}
+	}
+
+	if (fileBuildTags.length === 0) {
+		return;  // no build tags in file
+	}
+
+	let areInSetting = false;
+	const goConfig = vscode.workspace.getConfiguration('go', e.document.uri);
+	let settBldTags = goConfig['buildTags'];
+
+	if (settBldTags) {
+		const splChr = settBldTags.includes(',') ? ',' : ' ';
+		settBldTags = settBldTags.split(splChr);
+		areInSetting = fileBuildTags.every((tag: string) => {
+			// build tags OR array containing AND array
```

Comment:
I thought go.buildTags are for -tags flag. Usually architecture and os related build constraints are supposed to be specified using environment variables (GOOS/GOARCH) instead of -tags flag.

Reply:
You are right! There are many implicitly provided words like os and arch specified in GOOS & GOARCH. Also the compiler, CGO, version of Go and the "ignore" word. So only the additional words have to be satiesfied via -tags. ([https://golang.org/pkg/go/build/](url)).

So I would not check for: Currently supported os/arch, gc. gccgo, cgo, Go version gox.xx, "ignore" and still catch missing custom build tag like 'development'. 

Is this still wanted or does it smell like 'over engineering'?